packages/flutter_tools/lib/src/windows/windows_version_validator.dart

```diff
@@ -0,0 +1,88 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+import 'dart:io';
+
+import '../doctor_validator.dart';
+
+class WindowsVersionValidator extends DoctorValidator {
+  const WindowsVersionValidator() : super('Windows Version');
+  @override
+  Future<ValidationResult> validate() async {
+    final ProcessResult result = await Process.run('systeminfo', <String>[]);
+    final String resultStdout = result.stdout as String;
+
+    // Define the major versions that are not supported
+    const List<String> unsupportedVersions = <String>[
+      '7',
+      '8',
+    ];
+
+    final List<String> systemInfoElements = resultStdout.split('\n');
+
+    // Regular expression pattern for identifying
+    // semantic versioned strings
+    // (ie. 10.5.4123)
+    final RegExp regex = RegExp(r'^([0-9]+)\.([0-9]+)\.([0-9]+)$');
+
+    // Define the list that will contain the matches;
+    // if ran successfully, this list should have only
+    // one item
+    final List<String> versionList = <String>[];
+
+    // Use two booleans to identify when you have found
+    // the word 'version' and a version number that matches
+    // the regex pattern above; only once both are found do
+    // we report back a valid version
+    bool versionText = false;
+    bool versionSemver = false;
+    String? version;
+    for (int i = 0; i < systemInfoElements.length; i++) {
```

Comment:
nit: you can iterate directly over the `systemInfoElements` (as well as the `lineElements`) and avoid using `.elementAt`

```
for (final String curLine in systemInfoElements) {...}
```

Reply:
nice, i meant to actually refactor this to use that notation